components/InlineEdit.js

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+import styles from '../styles/InlineEdit.module.css'
+
+export default function InlineEdit(props) {
+  const { value, setValue, handleSubmit } = props;
+  const [editingValue, setEditingValue] = useState(value);
+  
+  const onChange = (event) => setEditingValue(event.target.value);
+  
+  const onKeyDown = (event) => {
+    if (event.key === "Enter" && "handleSubmit" in props) {
+      handleSubmit(editingValue);
+      event.target.blur();
+    }
+    if (event.key === "Escape" || event.key === "Enter") event.target.blur();
+  }
+  
+  const onBlur = (event) => {
+    if (event.target.value.trim() === "") {
+      setEditingValue(value);
+    } else {
+      setEditingValue(event.target.value)
+    }
```

Comment:
can you explain what these lines are doing?

Reply:
If the input box is empty after being trimmed for whitespace, 

> Just reviewed the last commit ([9766205](https://github.com/farhaannishtar/Habit-Tracker/commit/97662056f897f9339448ed88214e990e15dd5337)) but it looks so much better than what you had yesterday! Both aesthetically, and the code is so much cleaner too.
> 
> Love that the InlineEdit is it's own component. Did you find the code for that online, or write it yourself?
> 
> Regardless, this is looking a lot better!
> 
> p.s. reminder for future tickets to make a new branch first, then commit new code. Then you can open up a PR against this branch.

I found the code for InlineEdit online. I just changed a few things.

Reply:
After the input box loses focus, (`onBlur`), if the input box is empty after being trimmed for whitespace, don't change the `editingValue` to `""`.  Else, change the editing value to the value of the input box